librosa/display.py

```diff
@@ -509,6 +509,56 @@ def __init__(self, times, y, steps, envelope, sr=22050, max_samples=11025):
         self.envelope = envelope
         self.sr = sr
         self.max_samples = max_samples
+        self.cid = None
+        self.ax = None
+
+    def __del__(self):
+        self.disconnect(strict=True)
+
+    def connect(self, ax, *, signal="xlim_changed"):
+        """Connect the adaptor to a signal on an axes object.
+
+        Note that if the adaptor has already been connected to an axes object,
+        that connect is first broken and then replaced by a new callback.
+
+        Parameters
+        ----------
+        ax : matplotlib.axes.Axes
+            The axes to connect with this adaptor's `update`
+        signal : string, {"xlim_changed", "ylim_changed"}
+            The signal to connect
+
+        See Also
+        --------
+        disconnect
+        matplotlib.axes.callbacks.connect
+        """
+        # Disconnect any existing callback first
+        self.disconnect()
+
+        # Attach to axes and store the connection id
+        self.ax = ax
+        self.cid = ax.callbacks.connect(signal, self.update)
+
+    def disconnect(self, *, strict=False):
+        """Disconnect the adaptor's update callback.
+
+        Parameters
+        ----------
+        strict : bool
+            If `True`, remove references to the connected axes.
```

Comment:
maybe explain the consequences of removing these references?

Reply:
There shouldn't really be any user-facing consequences.  This is just a safety measure to eliminate circular references when you call `__del__`.

Reply:
cool. i see it now. maybe say that?